File: T6/codigos2122/JSON/app/src/main/java/com/example/json/dialogos/DialogoInfo.java
package com.example.json.dialogos;

import android.app.Dialog;
import android.content.Context;
import android.content.DialogInterface;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;
import androidx.fragment.app.DialogFragment;

public class DialogoInfo extends DialogFragment {

    private String nombre;

    public static DialogoInfo newInstance(String nombre) {

        Bundle args = new Bundle();
        args.putString("nombre", nombre);
        DialogoInfo dialogo = new DialogoInfo();
        dialogo.setArguments(args);
        return dialogo;
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        nombre = getArguments().getString("nombre");
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {

        AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());

        builder.setMessage("Estas seguro que el" + nombre+ " es tu equipo favorito");
        builder.setPositiveButton("SI", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {

                
            }
        });
        builder.setNegativeButton("NO", new DialogInterface.OnClickListener() {
            @Override
            public void onClick(DialogInterface dialogInterface, int i) {

            }
        });
        return builder.create();
    }

    public interface OnDialogoConfirm{
        void onDialogoSelected();
    }
}

File: Kotlin/T5/codigos_ces/Dialogos/app/src/main/java/com/example/dialogos/ui/dialog/DialogoInfo.kt
package com.example.dialogos.ui.dialog

import android.app.Dialog
import android.content.Context
import android.content.DialogInterface
import android.os.Bundle
import androidx.appcompat.app.AlertDialog
import androidx.fragment.app.DialogFragment
import com.example.dialogos.MainActivity
import com.google.android.material.snackbar.Snackbar

class DialogoInfo : DialogFragment() {

    lateinit var listener: OnDialogoListener

    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = context as MainActivity

    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        var builder: AlertDialog.Builder = AlertDialog.Builder(requireActivity())
        builder.setTitle("Ejemplo de cuadro de dialogo")
        builder.setMessage("Cuadro de dialogo para la informacion del sistema")
        builder.setPositiveButton("OK", { _, _ ->
            listener.onDialogoSelected("Comunicacion correcta",1)
        })
        builder.setNegativeButton("NO", { _, _ ->
            //listener.onDialogoSelected("Comunicacion incorrecta",2)
            listener.onDialogoSelectedAll()
        })
        builder.setNeutralButton("Cancel", { _, _ ->
            dismiss()
        })
        return builder.create()
    }

    interface OnDialogoListener{
        fun onDialogoSelected(respuesta: String, respuestaNum: Int)
        fun onDialogoSelectedAll()
    }

}